aep/general/0143/aep.md.j2

```diff
@@ -0,0 +1,116 @@
+# Standardized codes
+
+Many common concepts, such as spoken languages, countries, currency, and so on,
+have common codes (usually formalized by the [International Organization for
+Standardization][iso]) that are used in data communication and processing.
+These codes address the issue that there are often different ways to express
+the same concept in written language (for example, "United States" and "USA",
+or "Español" and "Spanish").
+
+## Guidance
+
+For concepts where a standardized code exists and is in common use, fields
+representing these concepts **should** use the standardized code for both input
+and output.
+
+{% tab proto %}
+
+```proto
+// A message representing a book.
+message Book {
+  // Other fields...
+
+  // The IETF BCP-47 language code representing the language in which
+  // the book was originally written.
+  // https://en.wikipedia.org/wiki/IETF_language_tag
+  string language_code = 99;
+}
+```
+
+{% tab oas %}
+
+```json
+
+{
+  "type": "object",
+  "properties": {
+    "language_code": {
+      "type": "string",
+      "description": "The IETF BCP-47 language code representing the language in which the book was originally written."
+    }
+  }
+}
+
+```
+
+{% endtabs %}
+
+- Fields representing standardized concepts **must** use the appropriate data
+  type for the standard code (usually `string`).
+  - Fields representing standardized concepts **should not** use enums, even if
+    they only allow a small subset of possible values. Using enums in this
+    situation often leads to frustrating lookup tables when using multiple APIs
+    together.
+  - Fields representing standardized concepts **must** indicate which standard
+    they follow, preferably with a link (either to the standard itself, the
+    Wikipedia description, or something similar).
+- The field name **should** end in `_code` or `_type` unless the concept has an
+  obviously clearer suffix.
+- When accepting values provided by users, validation **should** be
+  case-insensitive unless this would introduce ambiguity (for example, accept
+  both `en-gb` and `en-GB`). When providing values to users, APIs **should**
+  use the canonical case (in the example above, `en-GB`).
+
+### Content types
+
+Fields representing a content or media type **must** use [IANA media types][].
+For legacy reasons, the field **should** be called `mime_type`.
```

Comment:
Are we constrained by legacy here?

Reply:
That's a good question.  Is the "legacy" in question regarding pre-AIP Google APIs, or is it more about the historical usage of "MIME type" more generally?

Reply:
I'm not entirely sure. I guess since HTTP calls is content-type, we should follow suit?

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type

I don't think we should be constrained by legacy - if there's backwards-incompatible changes we need to introduce let's just provide migration guides. I'd rather minimize the number of suboptimal changes we make for legacy, but open to opinions!